rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -0,0 +1,173 @@
+		if (dialog.open == Window.OK) {
+			val selectedProfile = dialog.result
+			return #{TclMavenLauncher.PROFILE -> selectedProfile.get(0)}
+		} else {
+			return null // cancelled
+		}
+	}
+
+	private def Iterable<String> mavenGetProfilesWithUiFeedback(IProject project) {
+		val Iterable<String>[] container = newArrayOfSize(1)
+		progressRunner.run([ monitor |
+			monitor.beginTask("Collect maven profiles", IProgressMonitor.UNKNOWN)
+			container.set(0, mavenLauncher.getProfiles(project))
+			monitor.done
+		])
+		return container.head
+	}
+
+	private def boolean launchTest(IStructuredSelection selection, IProject project, String elementId,
+		TclLauncher launcher, Map<String, Object> options) {
+		logger.info("Trying to launch launcherClass='{}' test execution for elementId='{}' in project='{}'",
+			launcher.class.simpleName, elementId, project)
+		progressRunner.run([ monitor |
+			monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
+			val result = launcher.launchTest(selection, project, elementId, monitor, options)
+			project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
+			val expectedFile = (result.get(TclLauncher.EXPECTED_FILE) as File)
+			if (expectedFile == null) {
+				logger.error("resulting expectedFile must not be null")
+			} else {
+				safeUpdateJunitTestView(elementId, expectedFile)
+			}
+			monitor.done
+		])
+		return true
+	}
+
+	/** 
+	 * provide test result file (either the one created by the test run, or, if absent/on error 
+	 * a default error file) that is imported into junit (and thus displayed) 
+	 * */
+	private def void safeUpdateJunitTestView(String elementId, File expectedFile) {
+		val parentFolder = new File(expectedFile.parent)
+		if (!parentFolder.exists && !parentFolder.mkdirs) {
+			logger.error('''failed to change into parentFolder='{}' ''', parentFolder.absolutePath)
+		} else {
+			val newFile = File.createTempFile(expectedFile.name, ".xml", parentFolder)
+			if (expectedFile.exists) {
+				var OutputStream os = null
+				try {
+					os = new FileOutputStream(newFile)
+					Files.copy(expectedFile.toPath, os)
+					Files.delete(expectedFile.toPath)
+				} catch (IOException e) {
+					logger.error("error during storage of test run result " + expectedFile.path, e)
+					writeErrorFile(elementId, newFile)
+				} finally {
+					try {
+						os?.close
+					} catch (IOException e) {
+						logger.
+							error('''error closing the outputstream during copy to newFile='«newFile.absolutePath»' ''',
+								e)
+					}
+				}
+			} else {
+				writeErrorFile(elementId, newFile)
+			}
+			JUnitCore.importTestRunSession(newFile)
+			try {
+				Files.delete(newFile.toPath)
+			} catch (IOException e) {
+				logger.warn("error during removal of obsolete test result " + newFile.path, e)
+			}
+		}
+	}
+
+	/**
+	 * write a default error file for junit
+	 */
+	private def void writeErrorFile(String elementId, File file) {
+		try {
+			val os = new FileOutputStream(file)
```

Comment:
We should have Guava in the taget platform. There is no need to manually write files.